src/Managers/FileManager.php

```diff
@@ -84,14 +94,15 @@ private function getPhpFilesFromPath(string $path)
      */
     private function fileShouldBeIgnored(SplFileInfo $file): bool
     {
-        foreach ($this->filesToIgnore as $fileToIgnore) {
+        $path = $file->getPathname();
+
+        $matchingFilesToIgnore = array_filter($this->filesToIgnore, function (string $fileToIgnore) use ($path): bool {
             $regex = $this->patternToRegex($fileToIgnore);
-            if (preg_match("#{$regex}#", $file->getPathName())) {
-                return true;
```

Comment:
Too many indentation levels.

Unfortunately I haven't found a better solution, let me know if you can think of one.